doc/ref_arch/kubernetes/chapters/chapter03.md

```diff
@@ -209,70 +209,99 @@ infrastructure itself from the workloads compute resources.
+by the Kubernetes applications. This allows resources such as "GPUs,
+high-performance NICs, FPGAs, InfiniBand adapters, and other similar computing
+resources that may require vendor specific initialization and setup" to be
+managed and consumed via standard interfaces.
+
+Figure 3-2 below shows the main building blocks of a Kubernetes networking solution:
+- **Kubernetes Control Plane**: this is the core of a Kubernetes cluster - the
+apiserver, etcd cluster, kube-scheduler and the various controller-managers. The
+control plane (in particular the apiserver) provide a centralised point by which
+the networking solution is managed using an abstract management API.
+- **Default CNI Plugin (Cluster Network)**: this is the default cluster network plugin
+that has been deployed within the cluster to provide IP addresses to Pods. Note that
+support for IPv6 requires not only changes in the Kubernetes control plane, but
+also requires the use of a CNI Plugin that support dual-stack networking.
+- **CNI multiplexer/meta-plugin**: this is the component that integrates with
+the Kubernetes control plane via CNI, but allows for the use of multiple CNI
+plugins and the provision of multiple network connections to each Pod, as shown
+by the use of additional CNI Plugin and `net0` connection in the Pod. Note that
+the different network characteristics of the interfaces might require different
+networking technologies, which would potentially require different CNI plugins.
+Also note that this is only required for the Network Intensive profile.  Example
+CNI implementations which meet these requirements include Multus and DANM.  For
+more detailed feature comparison of CNI multiplexers/meta-plugins see Table 3-1 below.
+- **CNI Plugin (Additional)**: this is a CNI plugin that is used to provide
+additional networking needs to Pods, that aren't provided by the default CNI plugin.
+This can include connectivity to underlay networks via accelerated hardware devices.
+- **Device Plugin**: this is a Kubernetes extension that allows for the management
+and advertisement of vendor hardware devices. In particular, devices such as
+FPGA, SR-IOV NICs, SmartNICs, etc. can be made available to Pods by using Device Plugins.
+Note that alignment of these devices, CPU topology and Huge Pages will need the use
+of the [Topology Manager](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/).
+- **External / Application Load Balancing**: As Kubernetes Ingress, Egress and Services have no
+support for all the protocols needed in telecommunication environments
+(Diameter, SIP, LDAP, etc) and their capacity is limited, the architecture includes
+the use of alternative load balancers, including external or ones built into the
+application. Management of external load balancers must be possible via Kubernetes API objects.
+- **Other Features**: these additional features that are required by the
```

Comment:
I think we should list the "Other features" here and for the IPAM it should be noted, that the same IPAM should be responsible for the IP address assignment of all CNI plugins or other networking solutions.

Reply:
I'm not sure the requirement "requires" that.  

req.inf.ntw.10 states "The Architecture must support cluster wide coordination of IP address assignment." - I would argue that this doesn't necessarily mean a single IPAM component, as long as anything "doing" IPAM is aware of and communicates, somehow, with anything else doing IPAM - i.e. it is all co-ordinated.

Reply:
@CsatariGergely what are your thoughts?   I think either we need to change the requirement in chapter 2 to specify that a single IPAM component is required to manage cluster wide IP address assignment, or we accept that co-ordination between multiple IPAM components is acceptable.

Reply:
I don't necessarily disagree, but I don't see why all IPAM in a cluster needs to be centralised.  Does anyone have a concrete reason why this requirement is needed?  I am happy to restrict the scope to the default cluster network (as I don't see the problem in something else managing the IPAM for additional networking needs).

Reply:
@CsatariGergely updated in https://github.com/cntt-n/CNTT/pull/2121/commits/0bca1dbf3c64d3164a057ac4beff6909596fb01e - are you ok to resolve the conversation?